sdks/python/apache_beam/runners/portability/flink_runner.py

```diff
@@ -43,7 +43,13 @@ def default_job_server(self, options):
 class FlinkRunnerOptions(pipeline_options.PipelineOptions):
   @classmethod
   def _add_argparse_args(cls, parser):
-    parser.add_argument('--flink_master', default='[local]')
+    parser.add_argument('--flink_master',
+                        default='[auto]',
+                        help='Flink master address (host:port) to submit the'
+                             ' job against. Use "[local]" to start a local'
+                             ' cluster for the execution. Use "[auto]" if you'
+                             ' plan to either execute locally or submit through'
```

Comment:
How is this choice made if `[auto]` is passed? Should it really be the default?

Reply:
I'm coming from the Java SDK, where we use `[auto]` as the default. I think it makes sense because `[local]` has unexpected consequences when you use it with the Flink CLI (now that we have executable jars that is possible). It will force a local execution instead of submitting the job to the cluster. We could even think about removing the local mode, since it is covered by the auto mode as well.